[Feat] #3 - main menuUI - jason yang

VesselWheel opened the pull request:
<!--
  Assignees - 본인 선택하기
  ReviewWers - 팀원 선택하기
-->

## Contents
<!-- 작업 사항에 대한 설명을 적어주세요 --> 
 - collectinoView의 내부셀, 이미지, 레이블 여백 조정
 - HeaderView와 연결 시 매소드명 변경 ~Button() -> ~View()

## Screenshots
<!-- 작업 사항에 대한 스크린샷을 첨부해주세요 --> 
![issue1](https://github.com/NBCAMP-kiosk-team3/iOS/assets/88497383/6a9b8c26-6495-4cee-ac72-1b8fce3eacbf)



## To Reviewers
<!-- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이나 논의할 부분이 있다면 적어주세요 --> 
- MeinMenu.swift에서 71번째 행 주석처리한, 
// orderListView.tempOrderList.append(selectedData) 을 OrderViewList.swift에 데이터를 전송하려고하는데, SpabucksMenuItem 타입과 SpabucksOrderItem 타입이 달라, 통일해야할까요?


## Issues
<!-- 이슈가 발생하는 부분이 있다면 적어주세요 -->
protocol Eatable의 price 타입을 Int로 변경하는게 좋은가요 아니면 Double로 유지하고 소수점을 제외하는 함수를 사용할까요?

VesselWheel committed:
[Chore] #3 - CollectionView UI 조정 & MenuCell 클래스 분리
- collectinoView의 내부셀, 이미지, 레이블 여백 조정
- HeaderView와 연결 시 매소드명 변경 ~Button() -> ~View()
- didSelectItemAt 실행 시 데이터를 객체화
- issue :  dataSource의 타입과 OrderViewList의 tempOrderList의 타입 통일 필요

Yu-RiChoi commented on SPABUCKS-Kiosk-iOS/SPABUCKS-Kiosk-iOS/Presentation/Views/MenuVIew/MenuView.swift:
```diff
@@ -49,12 +49,12 @@ extension MenuView {
     }
-    func foodMenuButton() {
+    func foodMenuView() {
         dataSource = foodItems
         collectionView.reloadData()
     }
     
-    func mdMenuButton() {
+    func mdMenuView() {
         dataSource = merchandiserItems
         collectionView.reloadData()
     }
```
위에 showBeverageView처럼 showFoodView, showMdView 이렇게 수정해주시면 좋을 것 같습니다!

VesselWheel replied:
수정했어요 감사합니다!

yenny42 commented on SPABUCKS-Kiosk-iOS/SPABUCKS-Kiosk-iOS/Presentation/Views/MenuVIew/MenuView.swift:
```diff
@@ -7,35 +7,35 @@
-
-// MARK: - 메뉴 프로퍼티 정의
+    
+    // MARK: - 메뉴 프로퍼티 정의
```
p5; Mark 주석 형태를 통일하는 것도 괜찮을 것 같아요 :D

```
// MARK: - Properties
// MARK: - UI Properties
// MARK: - Life Cycle
// MARK: - Extensions
```
저는 현재 이 정도의 주석을 사용하고 있어요. ViewController에서 백준님이 적어두셨더라구요.
메뉴 프로퍼티 정의라는 것도 명확하게 알 수 있는 주석이지만, 저희가 컨벤션을 열심히 지키는 만큼 주석도 통일성을 가지면 깔끔해보이지 않을까? 하는 생각을 해봅니다ㅎㅎ

VesselWheel replied:
좋아요!

yenny42 commented on SPABUCKS-Kiosk-iOS/SPABUCKS-Kiosk-iOS/Presentation/Views/MenuVIew/MenuView.swift:
```diff
@@ -7,35 +7,35 @@
+        SpabucksMenuItem(id: 3, name: "Caffe Latte", imageName: "Caffe Latte", price: 8000)]
     
     var foodItems: [SpabucksMenuItem] = [
-        SpabucksMenuItem(id: 0, name: "Croissant", imageName: "croissant", price: 5.0),
-        SpabucksMenuItem(id: 1, name: "Sandwich", imageName: "sandwich", price: 6.0),
-        SpabucksMenuItem(id: 2, name: "Salad", imageName: "salad", price: 6.5),
-        SpabucksMenuItem(id: 3, name: "Scorn", imageName: "scorn", price: 5.5)]
+        SpabucksMenuItem(id: 0, name: "Croissant", imageName: "croissant", price: 5000),
+        SpabucksMenuItem(id: 1, name: "Sandwich", imageName: "sandwich", price: 6000),
+        SpabucksMenuItem(id: 2, name: "Salad", imageName: "salad", price: 6000),
+        SpabucksMenuItem(id: 3, name: "Scorn", imageName: "scorn", price: 5500)]
     
     var merchandiserItems: [SpabucksMenuItem] = [
-        SpabucksMenuItem(id: 0, name: "Tumbler", imageName: "Tumbler", price: 5.5),
-        SpabucksMenuItem(id: 1, name: "Pen", imageName: "Pen", price: 15.0),
-        SpabucksMenuItem(id: 2, name: "Notebook", imageName: "Notebook", price: 12.5),
-        SpabucksMenuItem(id: 3, name: "Tent", imageName: "Tent", price: 25.0)]
+        SpabucksMenuItem(id: 0, name: "Tumbler", imageName: "Tumbler", price: 5500),
+        SpabucksMenuItem(id: 1, name: "Pen", imageName: "Pen", price: 1500),
+        SpabucksMenuItem(id: 2, name: "Notebook", imageName: "Notebook", price: 12500),
+        SpabucksMenuItem(id: 3, name: "Tent", imageName: "Tent", price: 25000)]
```
p5; menuView 클래스는 메뉴 리스트를 그려주는 부분인데요, 데이터 소스를 다른 파일로 분리해서 사용하면 코드 모듈화 측면에서 괜찮을 것 같은데 한 번 생각해보셔도 좋을 것 같습니다 :D

VesselWheel committed:
[Chore] #3 - MenuCell 식별자, price 출력 값 수정
- MenuCell 식별자 : collectionViewCellIdentifier 으로 수정
- cell.priceLabel.text의 더블 소수점 제거
- ViewController 에서 사용된 show[메뉴]View 반영

yenny42 commented:
p3; 이번에 저희가 올린 풀리퀘스트들 전부 머지하고 브랜치를 닫으면 Main에 반영할 예정입니다. 사용하지 않는 죽은 주석들, 메모용 주석들 전부 삭제해주시면 좋을 것 같습니다 :)

yenny42 commented:
`protocol Eatable의 price 타입을 Int로 변경하는게 좋은가요 아니면 Double로 유지하고 소수점을 제외하는 함수를 사용할까요?`

둘 다 괜찮을 것 같은데요? 그런데 키오스크 특성상 소수점 아래로 내려가는 price 값이 계산되지 않을 것이기 때문에 Int로 지정하셔도 무방하다고 생각합니다 ㅎㅎ

VesselWheel committed:
[Chore] #3 - 주석 정리
